stdlib/Profile/src/Allocs.jl

```diff
@@ -78,6 +88,18 @@ Stop recording allocations.
 """
 function stop()
     ccall(:jl_stop_alloc_profile, Cvoid, ())
+
+    gc_num_after = Base.gc_num()
+    gc_diff = Base.GC_Diff(gc_num_after, g_gc_num[])
+    alloc_count = Base.gc_alloc_count(gc_diff)
+    g_total_allocs[] = g_total_allocs[] + alloc_count
+
+    # TODO: ugh, fetch has side effects
+    raw_results = ccall(:jl_fetch_alloc_profile, RawAllocResults, ())
+    num_sampled = raw_results.num_allocs
+    estimated_total = round(Int, Float64(num_sampled) / g_sample_rate[])
+
+    g_estimated_total_allocs[] = g_estimated_total_allocs[] + estimated_total
```

Comment:
I don't think you need to compute this here, right?

You can multiply the `alloc_count` by `g_sample_rate[]` instead of dividing the sampled allocs by the sample rate to get to the expected number of samples, and then compare that against the actual samples?

And then actually i think you don't need `g_total_allocs[]`, either, this takes the place of that. So something like:

```suggestion
    gc_num_after = Base.gc_num()
    gc_diff = Base.GC_Diff(gc_num_after, g_gc_num[])
    actual_alloc_count = Base.gc_alloc_count(gc_diff)

    expected_samples = round(Int, Float64(actual_alloc_count) * g_sample_rate[])

    g_expected_sampled_allocs[] += expected_samples
```

Then later, in `fetch()`, you can compare `length(raw_results.allocs)` with `g_expected_sampled_allocs[]` to get your hit rate?

And when `sample_rate[] = 1.0`, this should give you exactly how many allocations you missed.

Reply:
Oh yeah, good point 🧠

Reply:
done